plugins/inputs/sql/sql.go

```diff
@@ -0,0 +1,809 @@
+	}
+	if len(s.FieldTimestamp) > 0 && !match_str(s.FieldTimestamp, s.column_name) {
+		return fmt.Errorf("Missing column %s for given field_measurement", s.FieldTimestamp)
+	}
+	if len(s.FieldName) > 0 && !match_str(s.FieldName, s.column_name) {
+		return fmt.Errorf("Missing column %s for given field_measurement", s.FieldName)
+	}
+	if len(s.FieldValue) > 0 && !match_str(s.FieldValue, s.column_name) {
+		return fmt.Errorf("Missing column %s for given field_measurement", s.FieldValue)
+	}
+	if (len(s.FieldValue) > 0 && len(s.FieldName) == 0) || (len(s.FieldName) > 0 && len(s.FieldValue) == 0) {
+		return fmt.Errorf("Both field_name and field_value should be set")
+	}
+
+	// fill columns info
+	var cell interface{}
+	for i := 0; i < col_count; i++ {
+		dest_type := TYPE_AUTO
+		field_matched := true
+
+		if match_str(s.column_name[i], s.TagCols) {
+			field_matched = false
+			s.tag_idx[s.tag_count] = i
+			s.tag_count++
+			cell = new(string)
+		} else if match_str(s.column_name[i], s.IntFields) {
+			dest_type = TYPE_INT
+			cell = new(sql.RawBytes)
+		} else if match_str(s.column_name[i], s.FloatFields) {
+			dest_type = TYPE_FLOAT
+			cell = new(sql.RawBytes)
+		} else if match_str(s.column_name[i], s.TimeFields) {
+			dest_type = TYPE_TIME
+			cell = new(sql.RawBytes)
+		} else if match_str(s.column_name[i], s.BoolFields) {
+			dest_type = TYPE_BOOL
+			cell = new(sql.RawBytes)
+		} else if match_str(s.column_name[i], s.FieldCols) {
+			dest_type = TYPE_AUTO
+			cell = new(sql.RawBytes)
+		} else if !s.IgnoreOtherFields {
+			dest_type = TYPE_AUTO
+			cell = new(sql.RawBytes)
+		} else {
+			field_matched = false
+			cell = new(sql.RawBytes)
+			if Debug {
+				log.Printf("I! Skipped field %s", s.column_name[i])
+			}
+		}
+
+		if Debug && !field_matched {
+			log.Printf("I! Column %d '%s' dest type  %d", i, s.column_name[i], dest_type)
+		}
+
+		if s.column_name[i] == s.FieldHost {
+			s.field_host_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldDatabase {
+			s.field_database_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldMeasurement {
+			s.field_measurement_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldName {
+			s.field_name_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldValue {
+			s.field_value_idx = i
+			s.field_value_type = dest_type
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldTimestamp {
+			s.field_timestamp_idx = i
+			field_matched = false
+		}
+
+		if field_matched {
+			s.field_type[s.field_count] = dest_type
+			s.field_idx[s.field_count] = i
+			s.field_count++
+		}
+		s.cells[i] = cell
+		s.cell_refs[i] = &s.cells[i]
+	}
+
+	if Debug {
+		log.Printf("I! Query structure with %d tags and %d fields on %d columns...", s.tag_count, s.field_count, col_count)
+	}
+
+	return nil
+}
+
+func ConvertString(name string, cell interface{}) (string, bool) {
+	value, ok := cell.(string)
+	if !ok {
```

Comment:
this code I believe would be far more understandable as a switch statement.

```
  switch value := call.(type) {
  case string:
    return value, true
  case []byte:
    return string(barr), true
  case int64:
    return strconv.FormatInt(value, 10), true
  default:
    if Debug {
      log.Printf("W! converting '%s' type %s raw data '%s'", name, reflect.TypeOf(value).Kind(), fmt.Sprintf("%v", value))
    }
    return fmt.Sprintf("%v", value), true
  }
```

and once that happens, we can ask ourselves, why are we returning a boolean if its always true?

returning the boolean is fine but I'd make the default case actually return false and not attempt to convert the value.